File: Labs/Lab 8/Lab 8.java
public int coinChange(int[] coins, int amount) {
        int[] arr = new int[amount+1];

        Arrays.fill(arr, amount + 1); 

        arr[0] = 0;

        for (int coin : coins) {
            for (int i = coin; i <= amount; i++) {
                arr[i] = Math.min(arr[i], arr[i - coin] + 1);
            }
        }

        if(arr[amount] > amount){
            return -1;
        } else{
            return arr[amount];
        }
}

File: Labs/Lab 8/Lab 8.py
def coinChange(self, coins: List[int], amount: int) -> int:
        arr = [amount+1] * (amount+1)
        arr[0] = 0

        for i in range(len(coins)):
            for j in range(coins[i], amount+1):
                arr[j] = min(arr[j], arr[j-coins[i]] + 1);
        
        if arr[amount] > amount:
            return -1
        else:
            return arr[amount]
